gc/verbose/handler_standard/VerboseHandlerOutputStandard.cpp

```diff
@@ -446,17 +494,9 @@ MM_VerboseHandlerOutputStandard::handleScavengeEnd(J9HookInterface** hook, uintp
 void
 MM_VerboseHandlerOutputStandard::handleConcurrentEndInternal(J9HookInterface** hook, uintptr_t eventNum, void* eventData)
 {
-	/* convert event from concurrent-end to scavenge-end */
-	MM_ScavengeEndEvent scavengeEndEvent;
 	MM_ConcurrentPhaseEndEvent *event = (MM_ConcurrentPhaseEndEvent *)eventData;
-
-	scavengeEndEvent.currentThread = event->currentThread;
-	scavengeEndEvent.timestamp = event->timestamp;
-	scavengeEndEvent.eventid = event->eventid;
-	scavengeEndEvent.subSpace = NULL; //unknown info
-	scavengeEndEvent.cycleEnd = false;
-
-	handleScavengeEndNoLock(hook, J9HOOK_MM_PRIVATE_SCAVENGE_END, &scavengeEndEvent);
+	MM_ConcurrentPhaseStatsBase *stats = (MM_ConcurrentPhaseStatsBase *)event->concurrentStats;
+	stats->handleConcurrentEndInternal(hook, eventNum, eventData, this);
```

Comment:
Based on my comment from `ConcurrentPhaseStatsBase.hpp - handleConcurrentEndInternal`...
Let's branch out here, a branch for Scavenge Concurrent End and one for Mark, and create `handleConcurrentMarkEndInternal` & `handleConcurrentScavengeEndInternal`.

Reply:
To determine which to branch to, let's check eventID. Right now, Concurrent End is triggered with event: `J9HOOK_MM_PRIVATE_CONCURRENT_PHASE_END` for both ConcurrentGC and Scavenger, let's differentiate  between them. You can do `J9HOOK_MM_PRIVATE_CONCURRENT_MARK_PHASE_END` and `J9HOOK_MM_PRIVATE_CONCURRENT_SCAVENGE_PHASE_END`, register the event to the same handler `handleConcurrentEnd`, but when we come in here `handleConcurrentEndInternal` branch out